common/libraries/clients/postgresql/src/main/java/org/eclipse/dataspaceconnector/clients/postgresql/statement/BigDecimalArgumentHandler.java

```diff
@@ -0,0 +1,33 @@
+/*
+ *  Copyright (c) 2021 Daimler TSS GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Daimler TSS GmbH - Initial API and Implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.clients.postgresql.statement;
+
+import java.math.BigDecimal;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+public enum BigDecimalArgumentHandler implements ArgumentHandler {
+    INSTANCE;
```

Comment:
Is an enum containing one element necessary?

Reply:
Those `ArgumentHandler`s are stateless so predestined for being singletons - and this is the shortest way to define a singleton in Java known to me. So no need for 
```java
    public static final Singleton INSTANCE = new Singleton();
    private Singleton() {}
    public static Singleton getInstance(){ return INSTANCE; }
```
nor any of its derivatives - just defining an enum :)
But, if you like, I can rewrite this into something else.

Reply:
Maybe, since the implementation methods are all one-liner, we could implement a single enum with all the implementations inside, doing that we won't need to list every implementation in `SimplePostgresqlClient` because we will need only to call the static method `values()` on the enum class.